src/os/bluestore/BlueFS.cc

```diff
@@ -1399,6 +1426,19 @@ int BlueFS::open_for_write(
     log_t.op_dir_link(dirname, filename, file->fnode.ino);
 
   *h = _create_writer(file);
+
+  if (0 == filename.compare(filename.length() - 4, 4, ".log")) {
+    (*h)->writer_type = BlueFS::WRITER_WAL;
+    if (logger) {
+      logger->inc(l_bluefs_files_written_wal);
+    }
+  } else if (0 == filename.compare(filename.length() - 4, 4, ".sst")) {
+    (*h)->writer_type = BlueFS::WRITER_SST;
+    if (logger) {
+      logger->inc(l_bluefs_files_written_sst);
+    }
+  }
+
```

Comment:
Better to have a else case to handle any future extensions or unit tests to create any file.